src/invocation/InvocationService.ts

```diff
@@ -8,19 +8,27 @@ import Address = require('../Address');
 import ExceptionCodec = require('../codec/ExceptionCodec');
 import {BitsUtil} from '../BitsUtil';
 
-export interface Invocation {
+var EXCEPTION_MESSAGE_TYPE = 109;
+var INVOCATION_TIMEOUT = 120000;
+var INVOCATION_RETRY_DELAY = 1000;
+
+class Invocation {
+
+    constructor(request: ClientMessage) {
+        this.request = request;
+    }
+
     request: ClientMessage;
-    partitionId?: number;
-    address?: Address;
-    connection?: ClientConnection;
-    deferred?: Q.Deferred<ClientMessage>;
-    handler?: (...args: any[]) => any;
+    partitionId: number;
+    address: Address;
+    deadline: Date = new Date(new Date().getTime() + INVOCATION_TIMEOUT);
```

Comment:
Ok. I created an issue so we do not forget it. https://github.com/hazelcast/hazelcast-nodejs-client/issues/52